src/robusta/core/sinks/robusta/robusta_sink.py

```diff
@@ -324,6 +333,56 @@ def __send_helm_release_events(self, release_data: List[HelmRelease]):
         except Exception:
             logging.error("Error occurred while sending `helm release` trigger event", exc_info=True)
 
+    def __discover_custom_namespaced_resources(self):
+        self.__assert_namespaces_cache_initialized()
+
+        logging.info(f"Starting namespace discovery loop every {NAMESPACE_DISCOVERY_SECONDS} seconds")
+
+        while True:
+            updated_namespaces: defaultdict = defaultdict(list)
+
+            try:
+                for resource in self.namespace_monitored_resources:
+                    try:
+                        results = Discovery.count_resources(
+                            kind=resource.kind,
+                            api_group=resource.apiGroup,
+                            version=resource.apiVersion
+                        )
+
+                        for namespace_name, count in results.items():
+                            updated_namespaces[namespace_name].append(ResourceCount(
+                                kind=resource.kind,
+                                apiVersion=resource.apiVersion,
+                                apiGroup=resource.apiGroup,
+                                count=count
+                            ))
+
+                    except ResourceAccessForbiddenError as e:
+                        logging.warning(f"Skipping resource {resource.kind} due to insufficient permissions: {e}")
+                    except Exception as e:
+                        logging.exception(f"Unexpected error counting resource {resource.kind}: {e}")
+
+                namespaces_to_publish = []
+                for namespace_name, resources in updated_namespaces.items():
+                    namespace_info = self.__namespaces_cache.get(namespace_name)
+                    if namespace_info:
+                        namespace_info.metadata = NamespaceMetadata(resources=resources)
+                        namespaces_to_publish.append(namespace_info)
+                    else:
+                        logging.warning(f"Namespace '{namespace_name}' not found in cache when assigning metadata.")
+
+                if namespaces_to_publish:
+                    self.dal.publish_namespaces(namespaces_to_publish)
+                    logging.info(f"Published metadata for {len(namespaces_to_publish)} namespaces")
+
+            except Exception as e:
+                logging.exception(f"Discovery process failed: {e}")
+
+            logging.info(f"Discovery completed, sleeping for {NAMESPACE_DISCOVERY_SECONDS} seconds")
```

Comment:
would probably be better to add it as a parameter on the sync, with a default value than env var